queries.sql

```diff
@@ -5,4 +5,110 @@ SELECT date_of_birth FROM animals WHERE name='Agumon' OR name='Pikachu';
 SELECT name,escape_attempts FROM animals WHERE weight_kg > 10.5;
 SELECT * FROM animals WHERE neutered = true;
 SELECT * FROM animals WHERE NOT name='Gabumon';
-SELECT * from animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
\ No newline at end of file
+SELECT * from animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+BEGIN;
+UPDATE
+  animals
+SET
+  species = 'unspecified';
+SELECT
+  *
+FROM
+  animals;
+ROLLBACK;
+SELECT
+  *
+FROM
+  animals;
+
+BEGIN;
+UPDATE
+  animals
+SET
+  species = 'digimon'
+WHERE
+  name LIKE '%mon';
+UPDATE
+  animals
+SET
+  species = 'pokemon'
+WHERE
+  species IS NULL;
+SELECT
+  *
+FROM
+  animals;
+COMMIT;
+SELECT
+  *
+FROM
+  animals;
+
+  BEGIN;
+DELETE FROM
+  animals;
+ROLLBACK;
+SELECT
+  *
+FROM
+  animals;
+
+BEGIN;
+DELETE FROM
+  animals
+WHERE
+  date_of_birth > '2022-01-01';
+SAVEPOINT SP1;
+UPDATE
+  animals
+SET
+  weight_kg = weight_kg * -1;
+ROLLBACK TO SP1;
+UPDATE
+  animals
+SET
+  weight_kg = weight_kg * -1
+WHERE
+  weight_kg < 0;
+COMMIT;
+
+SELECT
+  COUNT(*)
+FROM
+  animals;
+SELECT
+  COUNT(*)
+FROM
+  animals
+WHERE
+  escape_attempts = 0;
+SELECT
+  AVG(weight_kg)
+FROM
+  animals;
+SELECT
+  neutered,
+  MAX(escape_attempts)
+FROM
+  animals
+GROUP BY
+  neutered;
+SELECT
+  species,
+  MIN(weight_kg),
+  MAX(weight_kg)
+FROM
+  animals
+GROUP BY
+  species;
+SELECT
+  species,
+  AVG(escape_attempts)
+FROM
+  animals
+WHERE
+  date_of_birth BETWEEN '1990-01-01'
+  AND '2000-01-01'
+GROUP BY
```

Comment:
Kindly make sure that the selection includes the whole year of `2000`. you are only selecting up to 1st January 2000 and therefore excluding the days and months from 2nd January to December 31, 2000.

- The date selection range should be up to `2000/12/31`